wazo_dird/plugins/google_backend/services.py

```diff
@@ -218,37 +242,39 @@ def _extract_numbers_except_label(cls, contact):
 
         return numbers
 
+    @classmethod
+    def _find_name(cls, contact):
+        for name_obj in contact.get('names', []):
+            return name_obj
+        return {}
+
     @classmethod
     def _extract_name(cls, contact):
-        name = contact.get('gd$name', {}).get('gd$fullName', {}).get('$t', '')
+        name_obj = cls._find_name(contact)
+        name = name_obj.get('displayName', '')
         if not name:
-            name = contact.get('title', {}).get('$t', '')
+            name = name_obj.get('unstructuredName', '')
         return name
 
     @classmethod
     def _extract_first_name(cls, contact):
-        return contact.get('gd$name', {}).get('gd$givenName', {}).get('$t', '')
+        return cls._find_name(contact).get('givenName', '')
 
     @classmethod
     def _extract_last_name(cls, contact):
-        return contact.get('gd$name', {}).get('gd$familyName', {}).get('$t', '')
+        return cls._find_name(contact).get('familyName', '')
 
     @classmethod
     def _extract_type(cls, entry):
-        rel = entry.get('rel')
-        if rel:
-            _, type_ = rel.rsplit('#', 1)
-        else:
-            type_ = entry.get('label')
-        return type_
+        return entry.get('type', 'other')
```

Comment:
Not sure if this should stay like this, or move the 'other' into `extract_numbers_by_label()`

Reply:
This does seem to fix [WAZO-2439](https://wazo-dev.atlassian.net/browse/WAZO-2439)